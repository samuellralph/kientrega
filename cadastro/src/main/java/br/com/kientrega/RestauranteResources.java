package br.com.kientrega;

import br.com.kientrega.entidades.Restuarante;
import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.transaction.Transactional;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;
import java.util.Optional;

@Path("/restaurantes")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class RestauranteResources {

    @GET
    public List<Restuarante> buscarTodos (){
        return Restuarante.listAll();
    }

    @GET
    @Path("/{id}")
    public Restuarante buscarRestuarante (@PathParam("id") Long id){
        Optional<PanacheEntityBase> restauranteOp = Restuarante.findByIdOptional(id);
                Optional.ofNullable(restauranteOp).orElseThrow(() -> new NotFoundException("Restaurante não encontrado!"));
        return  (Restuarante) restauranteOp.get();
    }

    @POST
    @Transactional
    public  void adicionar(Restuarante dto) {
        dto.persist();
        Response.status(Response.Status.CREATED);
    }

    @PUT
    @Path("/{id}")
    @Transactional
    public  void alterar(@PathParam("id") Long id, Restuarante dto) {
        Optional<PanacheEntityBase> restauranteOp = Restuarante.findByIdOptional(id);
        Optional.ofNullable(restauranteOp).orElseThrow(() -> new NotFoundException("Restaurante não encontrado!"));
        Restuarante restaurante = (Restuarante) restauranteOp.get();
        restaurante.nome = dto.nome;
        restaurante.proprietario = dto.proprietario;
        restaurante.persist();
    }

}
