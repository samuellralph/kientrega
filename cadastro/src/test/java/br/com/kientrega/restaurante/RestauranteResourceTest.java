package br.com.kientrega.restaurante;

import br.com.kientrega.CadastroTestLifeCycleManager;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.codehaus.plexus.classworlds.strategy.Strategy;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;

import static io.restassured.RestAssured.given;


@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurante-cenario-1.yml")
    public void testBuscarRestaurantes(){
        //Assertions.assertEquals(1 , BigDecimal.ONE.intValue());
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }
}
