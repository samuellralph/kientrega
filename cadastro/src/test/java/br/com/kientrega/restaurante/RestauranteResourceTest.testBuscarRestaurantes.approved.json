[
  {
    "id": 123,
    "nome": "Novo proprietario",
    "pratos": [],
    "proprietario": "ID do Keycloak"
  },
  {
    "id": 234,
    "nome": "Novo proprietario",
    "pratos": [],
    "proprietario": "ID do Keycloak"
  }
]